### PR TITLE
Implement CQL ConvertsToDate and ConvertsToDateTime and add new query…

### PR DESCRIPTION
**Pull request by marlenaMyr**
… to test SocialSecurityNumber with ConvertsToInteger

### CODE REVIEW

**Review by alexanderkiel — CHANGES_REQUESTED** [review]

File: docs/conformance/cql.md [hunk 1]
```diff
@@ -322,38 +322,38 @@ The section numbers refer to the documentation of the [ELM Specification](https:
...
-| 22.23. | ToDateTime         | ✓          |               |       |
-| 22.24. | ToDecimal          | ✓          |               |       |
-| 22.25. | ToInteger          | ✓          |               |       |
-| 22.26. | ToList             | ✓          |               |       |
-| 22.27. | ToLong             | ✓          |               |       |
-| 22.28. | ToQuantity         | ✓          |               |       |
-| 22.29. | ToRatio            | ✗          |               |       |
-| 22.30. | ToString           | ✓          |               |       |
-| 22.31. | ToTime             | ✗          |               |       |
+|--------|--------------------|--|---------------|-------|
```
[the remark this answers is not in the document]
Comment (marlenaMyr) [reply docs/conformance/cql.md]: I have adjusted the table.

File: modules/cql/src/blaze/elm/compiler/type_operators.clj [hunk 2]
```diff
@@ -130,9 +130,41 @@
...
 
 
-;; TODO 22.8. ConvertsToDate
+;; 22.8. ConvertsToDate
+(defrecord ConvertsToDateOperatorExpression [operand]
+  core/Expression
+  (-eval [_ {:keys [now] :as context} resource scope]
+    (when-let [operand (core/-eval operand context resource scope)]
+      (when (some? operand)
+        (some? (p/to-date operand now))))))
```
[the remark this answers is not in the document]
Comment (marlenaMyr) [reply modules/cql/src/blaze/elm/compiler/type_operators.clj]: I added the form.

File: modules/cql/src/blaze/elm/compiler/type_operators.clj [hunk 3]
```diff
@@ -130,9 +130,41 @@
...
+                               now)))))
+  (-form [_]
+    (list 'converts-to-date-time (core/-form operand))))
+
+
+(defmethod core/compile* :elm.compiler.type/converts-to-date-time
+  [context {:keys [operand]}]
+  (when-let [operand (core/compile* context operand)]
+    (if (system/date? operand)
+      (when (some? operand)
```
[the remark this answers is not in the document]
Comment (marlenaMyr) [reply modules/cql/src/blaze/elm/compiler/type_operators.clj]: Yes you are right! I have removed the unnecessary code.

File: modules/cql/src/blaze/elm/compiler/type_operators.clj [hunk 4]
```diff
@@ -130,9 +130,41 @@
...
+    (->ConvertsToDateOperatorExpression operand)))
+
+
+;; 22.9. ConvertsToDateTime
+(defrecord ConvertsToDateTimeOperatorExpression [operand]
+  core/Expression
+  (-eval [_ {:keys [now] :as context} resource scope]
+    (when-let [operand (core/-eval operand context resource scope)]
+      (when (some? operand)
+        (some? (p/to-date-time operand
```
[the remark this answers is not in the document]
Comment (marlenaMyr) [reply modules/cql/src/blaze/elm/compiler/type_operators.clj]: Done.

File: modules/cql/src/blaze/elm/compiler/type_operators.clj [hunk 5]
```diff
@@ -130,9 +130,41 @@
...
 
-;; TODO 22.8. ConvertsToDate
+;; 22.8. ConvertsToDate
+(defrecord ConvertsToDateOperatorExpression [operand]
+  core/Expression
+  (-eval [_ {:keys [now] :as context} resource scope]
+    (when-let [operand (core/-eval operand context resource scope)]
+      (when (some? operand)
+        (some? (p/to-date operand now)))))
+  (-form [_]
```
Comment (alexanderkiel) [inline modules/cql/src/blaze/elm/compiler/type_operators.clj:140]: The test for `form` is missing.

File: modules/cql/src/blaze/elm/compiler/type_operators.clj (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexanderkiel) [reply modules/cql/src/blaze/elm/compiler/type_operators.clj]: The test is missing.

**Review by alexanderkiel — APPROVED** [review]

**Pull request merged**